samigo/samigo-qti/src/java/org/sakaiproject/tool/assessment/qti/helper/AuthoringHelper.java

```diff
@@ -679,43 +697,205 @@ public AssessmentFacade createImportedAssessment(Document document, String unzip
+            section.addSectionMetaData(SectionDataIfc.POOLID_FOR_RANDOM_DRAW, pool.getQuestionPoolId().toString());
+            
+            Iterator iter = section.getItemFacadeSet().iterator();
+            HashSet questionPoolItems = new HashSet();
+            while (iter.hasNext()) {
+              ItemFacade item = (ItemFacade)iter.next();
+              
+              //copy item so we can have it in more than one assessment
+              ItemData item2 = itemService.cloneItem(item);
+              item2.setItemId(new Long(0));
+              item2.setSection(null);
+              item2.setSequence(null);
+              ItemFacade itemFacade = itemService.saveItem(new ItemFacade(item2));
+              //item.setItemId(item.getItemId());
+              //item.setItemIdString(item.getItemId().toString());
+              
+              QuestionPoolItemData questionPoolItem = new QuestionPoolItemData();
+              questionPoolItem.setQuestionPoolId(pool.getQuestionPoolId());
+              questionPoolItem.setItemId(itemFacade.getItemId());        
+              pool.addQuestionPoolItem((QuestionPoolItemIfc) questionPoolItem);
+            }
+            questionPoolService.savePool(pool);
           }
-          // make sure required fields are set
-          item.setCreatedBy(me);
-          item.setCreatedDate(assessment.getCreatedDate());
-          item.setLastModifiedBy(me);
-          item.setLastModifiedDate(assessment.getCreatedDate());
-          item.setStatus(ItemDataIfc.ACTIVE_STATUS);
-          // assign the next sequence number
-          item.setSequence(  Integer.valueOf(itm + 1));
-          // add item to section
-          item.setSection(section); // one to many
-          // update metadata with PARTID
-          item.addItemMetaData(ItemMetaData.PARTID, section.getSectionId().toString());
-           
-          // Item Attachment
-          exHelper.makeItemAttachmentSet(item);
           
-          item = itemService.saveItem(item);
-          section.addItem(item);
-          EventTrackingService.post(EventTrackingService.newEvent(SamigoConstants.EVENT_ASSESSMENT_SAVEITEM, "/sam/" + AgentFacade.getCurrentSiteId() + "/saved itemId=" + item.getItemId().toString(), true));
-        } // ... end for each item
-        
+        } // This assessment is not based on questionpool
+        else {
+          
+          List itemList = exHelper.getItemXmlList(sectionXml);
+          for (int itm = 0; itm < itemList.size(); itm++) // for each item
+          {
+            log.debug("items=" + itemList.size());
+            Item itemXml = (Item) itemList.get(itm);
+            
+            ItemFacade item = new ItemFacade();
+          try {
+            exHelper.updateItem(item, itemXml, isRespondus);
+          }
+          catch (RespondusMatchingException rme) {
+            if (failedMatchingQuestions != null) {
+              failedMatchingQuestions.add(itm + 1);
+            }
+          }
+            
```

Comment:
This piece of code (more or less) is very similar to a previous one you add, could they be moved to a common method?

Reply:
I prefer to stick to a tested version, if we have differences in 22.x and master we always need to compare, it happened in the past, it's better to use tested logic.